backend/de.metas.business/src/main/sql/postgresql/ddl/functions/Reset_System_Base_Language.sql

```diff
@@ -0,0 +1,82 @@
+CREATE OR REPLACE FUNCTION Reset_System_Base_Language(p_ad_language_id numeric = -1) RETURNS VOID
+AS
+$$
+DECLARE
+    destination_language  varchar := '';
+    source_language       varchar := '';
+    table_name            varchar;
+    base_table_id         numeric;
+    base_table            varchar;
+    query_set_template    text    := '';
+    final_update_query    text    := '';
+    v_columnname          varchar;
+    update_query_template varchar := 'WITH X AS (SELECT * FROM base_table_trl WHERE base_table_id = %s AND ad_language =  ''%s'') '
+                                         || 'UPDATE base_table as target '
+                                         || 'SET %s '
+                                         || 'FROM X '
+        || 'WHERE target.base_table_id = x.base_table_id ;' ;
+BEGIN
+
+    IF (p_ad_language_id IS NULL OR p_ad_language_id <= 0) THEN
+        RAISE WARNING 'No language is provided';
+        RETURN;
+    END IF;
+
+    SELECT l.ad_language INTO source_language FROM ad_language l WHERE l.isbaselanguage = 'Y';
+    SELECT l.ad_language INTO destination_language FROM ad_language l WHERE l.ad_language_id = p_ad_language_id;
+
+    IF (source_language IS NULL OR TRIM(source_language) = '') THEN
+        RAISE WARNING 'No source language is provided';
```

Comment:
use RAISE EXCEPTION

Reply:
error message shall be something like: `Previous base AD_Language was not found`.